# **TUGAS AKHIR DATA MINING**

*   Nama	: Khoirizqi Velarati
*   NIM	  : A11.2021.13895
*   Kelas	: A11.4404


> Note: Sebelumnya, saya memilih judul *Penerapan Data Mining untuk Prediksi Kebakaran Hutan Menggunakan Algoritma Decision Tree C4.5*.
Akan tetapi, setelah melakukan riset lebih dalam, saya memahami bahwa algoritma Decision Tree C4.5 tidak dapat saya gunakan pada data yang sudah saya pilih karena pada data tersebut, targer merupakan data numerik. Sedangkan untuk algoritma Decision Tree C4.5 dirancang untuk data kategoori atau kelas diskrit.
Kemudian, saya akhirnya mengganti algoritma menjadi algoritma Linear Regression.

## **PENERAPAN DATA MINING UNTUK PREDIKSI LUASAN AREA KEBAKARAN HUTAN MENGGUNAKAN ALGORITMA LINEAR REGRESSION**

> Menurut Gujarati pada tahun 2009, Linear Regression adalah metode pendekatan sebagai kajian terhadap ketergantungan satu variabel, yaitu antara variabel terikan Y dan satu atau lebih variabel bebas yang disebut X


---
### Deskripsi
*   Kebakaran hutan adalah salah satu masalah besar yang ada di Indonesia dan mengancam dunia.
*   Dengan dilakukannya prediksi luasan area ini, diharapkan petugas pemadam mendapatkan informasi terkait sumber daya yang diperlukan untuk memadamkan api














In [188]:
#Import Library
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

In [189]:
#Load Data
dataset = pd.read_csv('/content/drive/MyDrive/forest_fires.csv')
dataset

X  Y month  day  FFMC    DMC     DC   ISI  temp  RH  wind  rain   area
0    7  5   mar  fri  86.2   26.2   94.3   5.1   8.2  51   6.7   0.0   0.00
1    7  4   oct  tue  90.6   35.4  669.1   6.7  18.0  33   0.9   0.0   0.00
2    7  4   oct  sat  90.6   43.7  686.9   6.7  14.6  33   1.3   0.0   0.00
3    8  6   mar  fri  91.7   33.3   77.5   9.0   8.3  97   4.0   0.2   0.00
4    8  6   mar  sun  89.3   51.3  102.2   9.6  11.4  99   1.8   0.0   0.00
..  .. ..   ...  ...   ...    ...    ...   ...   ...  ..   ...   ...    ...
512  4  3   aug  sun  81.6   56.7  665.6   1.9  27.8  32   2.7   0.0   6.44
513  2  4   aug  sun  81.6   56.7  665.6   1.9  21.9  71   5.8   0.0  54.29
514  7  4   aug  sun  81.6   56.7  665.6   1.9  21.2  70   6.7   0.0  11.16
515  1  4   aug  sat  94.4  146.0  614.7  11.3  25.6  42   4.0   0.0   0.00
516  6  3   nov  tue  79.5    3.0  106.7   1.1  11.8  31   4.5   0.0   0.00

[517 rows x 13 columns]

In [190]:
#Hapus data yang tidak dibutuhkan
dataset = dataset.drop(columns=['X','Y','month','day'])

In [191]:
#Ambil sampel data
dataset.sample()

FFMC    DMC     DC   ISI  temp  RH  wind  rain  area
121  91.5  145.4  608.2  10.7  17.1  43   5.4   0.0   0.0

Note:
*   FFMC: indeks kelembaban bahan bakar halus
*   DMC:  indeks kelembaban tumpukan dedaunan
*   DC: indeks kekeringan
*   ISI: indeks penyebaran awan
*   temp: suhu udara
*   RH: persentase kelembaban udara relatif
*   wind: kecepatan angin
*   rain: curah hujan
*   area: luasan area





In [192]:
#Cek missing value
print(dataset.isnull().sum())

FFMC    0
DMC     0
DC      0
ISI     0
temp    0
RH      0
wind    0
rain    0
area    0
dtype: int64


Note:
*   Tidak ditemukan adanya missing value



In [193]:
#Cek tipe data
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 517 entries, 0 to 516
Data columns (total 9 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   FFMC    517 non-null    float64
 1   DMC     517 non-null    float64
 2   DC      517 non-null    float64
 3   ISI     517 non-null    float64
 4   temp    517 non-null    float64
 5   RH      517 non-null    int64  
 6   wind    517 non-null    float64
 7   rain    517 non-null    float64
 8   area    517 non-null    float64
dtypes: float64(8), int64(1)
memory usage: 36.5 KB


Note:
*   Data terdiri dari 517 baris dan 9 kolom
*   Semua data sudah bertipe numeric

In [194]:
# Split data
x = dataset.drop(columns='area').values
y = dataset['area'].values

Note:


> x merupakan variabel bebas, dan y merupakan variabel terikat



In [195]:
#Membagi data menjadi data latihan dan data pengujian
xtrain,xtest,ytrain,ytest = train_test_split(x,y,test_size=0.2,random_state=43)

In [196]:
#Membuat Model
lr = LinearRegression()

In [197]:
#melatih model menggunakan data latih.
lr.fit(x_train, y_train)

LinearRegression()

Note:


*   x_train adalah fitur-fitur dari data latih
*   y_train adalah target atau nilai yang sesungguhnya dari data latih.




In [198]:
#Prediksi pada data uji (x_test) menggunakan model yang telah terlatih.
y_pred = lr.predict(x_test)

In [199]:
#Menghitung skor akurasi
lr.score(x_test, y_test)

0.016721603785082362

Note:


> Didapatkan nilai akurasi sebesar 0.016721603785082362



In [200]:
#Menghitung nilai MSE
mean_squared_error(y_test, y_pred)

5344.518366381306

Note:


> Didapatkan nilai MSE sebesar 5344.518366381306

In [201]:
dataset.sample()

FFMC    DMC     DC  ISI  temp  RH  wind  rain  area
53  92.1  111.2  654.1  9.6  20.4  42   4.9   0.0   0.0

Note:


> Ambil data sampel di atas sebagai contoh data ketika melakukan prediksi



In [206]:
#prediksi
lr.predict([[92.1,	111.2,	654.1,	9.6,	20.4,	42,	4.9,	0.0	]])

array([14.69257129])

Note:


> Didapatkan nilai prediksi sebesar 14.69257129
